include/openPMD/auxiliary/Mpi.hpp

```diff
@@ -0,0 +1,145 @@
+/* Copyright 2022 Franz Poeschel
+ *
+ * This file is part of openPMD-api.
+ *
+ * openPMD-api is free software: you can redistribute it and/or modify
+ * it under the terms of of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * openPMD-api is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with openPMD-api.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include "openPMD/config.hpp"
+
+#include <optional>
+#include <stdexcept>
+#include <type_traits>
+#include <utility>
+
+#if openPMD_HAVE_MPI
+#include <mpi.h>
+#endif
+
+namespace openPMD::auxiliary
+{
+#if openPMD_HAVE_MPI
+
+using Mock_MPI_Comm = MPI_Comm;
```

Comment:
As discussed: Mocking in shipped code is dangerous and the reason we have to do a full linker-wrapping for ADIOS1.

Let's avoid this if we can write `#ifdef`s instead.

Reply:
I've removed these changes now.
The lines that are now still in the new `Mpi.hpp` file are moved there from `Filesystem.cpp`, I think it makes sense to keep that change.

Reply:
Ok, but the line `using Mock_MPI_Comm = MPI_Comm;` still needs to be cleaned out pls :)
```suggestion
```